scanner_test.go

```diff
@@ -353,6 +353,64 @@ func TestScan(t *testing.T) {
 			),
 		},
 
+		{
+			name: "oct null",
+			text: "\"\\0\"",
+			want: []byte{
+				0x00,
+			},
+		},
+		{
+			name: "oct null 8",
+			text: "\"\\08\"",
+			want: []byte{
+				0x00, 0x38,
+			},
+		},
+		{
+			name: "oct double",
+			text: "\"\\13\"",
+			want: []byte{
+				0x0b,
+			},
+		},
+		{
+			name: "oct double X",
+			text: "\"\\13" + "X\"",
+			want: []byte{
+				0x0b, 0x58,
+			},
+		},
+		{
+			name: "oct Y double",
+			text: "\"Y" + "\\13\"",
+			want: []byte{
+				0x59, 0x0b,
+			},
+		},
+		{
+			name: "oct triple",
+			text: "\"\\007\"",
+			want: []byte{
+				0x07,
+			},
+		},
+		{
+			name: "oct WoW",
+			text: "\"\\127o\\127\"",
+			want: []byte{
+				0x57, 0x6f, 0x57,
+			},
+		},
+
+		{
+			name: "oct hex oct",
+			text: "\"\\127\\x40\\127\"",
+			want: []byte{
+				0x57, 0x40, 0x57,
+			},
+		},
```

Comment:
Could you include tests for the following strings?
```go
"\1234"
"\8" // error
"\008" // error
"\0\0\0"
"\377"
"\400" // error, overflows a single byte
"\777" // error
```
You can make a "test for error" test by omitting `want`.

Reply:
"\008" isn't an error, it is a null followed by a '8'. But otherwise, done.

Reply:
oh wow, good thing I caught that!